### DISCUSSION TITLE
Is edge_attr message passing implemented in PYG's HGTConv graph structure?

### DISCUSSION THREAD

**Discussion by MMMMMz** [pyg-team/pytorch_geometric#7505] [Q&A]
Is edge_attr message passing implemented in PYG's HGTConv graph structure?

**Comment by rusty1s** [comment]
Currently no :(

**Reply by MMMMMz** [reply]
But HGTConv can accept edge_attr values through SparseTensor type data. 
How is “edge_attr” specifically used in HGTConv?

**Reply by rusty1s** [reply]
It is not used at all. If you set a value through `SparseTensor`, these values will be ignored currently.

**Reply by MMMMMz** [reply]
Thank you for your answer.Could you please confirm whether there is a heterogeneous graph structure available with edge_attr in PYG now?

**Reply by rusty1s** [reply]
For example, `HEATConv` supports it. More generally, both `HeteroConv` and `to_hetero` can take GNN layers with edge feature support in as well.
